### PR TITLE
Data_Engine: Set Name and Tag in AsCustomObject()

### LINKED ISSUE #2820: AsCustomObject() obliterates Name, Tags, and BHoM_Guid

### PR DESCRIPTION
**Pull request by JosefTaylor**
### Issues addressed by this PR

Closes #2820

### CODE REVIEW

**Comment by JosefTaylor** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

File: Data_Engine/Query/ValuesAt.cs [hunk 1]
```diff
@@ -123,7 +123,12 @@ private static CustomObject AsCustomObject(DataRow row, DataColumnCollection col
             CustomObject obj = new CustomObject();
             foreach (DataColumn col in columns)
             {
-                obj.CustomData[col.ColumnName] = row[col];
+                if (col.ColumnName == "Name")
+                    obj.Name = row[col].ToString();
+                if (col.ColumnName == "Tags")
```
Comment (IsakNaslundBh) [inline Data_Engine/Query/ValuesAt.cs:128]:
```suggestion
                else if (col.ColumnName == "Tags")
```

**Review by IsakNaslundBh — CHANGES_REQUESTED** [review]
Good catch!

Think this just need a slight adjustment as code currently adds the name to both the Name proeprty and to the Name key in the dictionary (without the `else if` on the tags, the `else` only is in relation to the Tags).

With that change I think this is good to go!

**Review by IsakNaslundBh — DISMISSED** [review]
Change makes sense to me. Tested and approved!

**Comment by JosefTaylor** [comment]
@BHoMBot check compliance
@BHoMBot check required

**Review by IsakNaslundBh — APPROVED** [review]
Tested, works great. Approved

**Pull request merged**